Refactor sitemaps

wojtekmach opened the pull request:
* Generate both hex & hexdocs sitemap through HexWeb.SitemapView
* Add test for hexdocs sitemap
* Unify markup between hex & hexdocs sitemap

wojtekmach commented:
I wasn't sure if there was a specific reason not to use templates for the docs sitemap, so I just went ahead and did that :)

lexmag commented on test/fixtures/sitemap.xml:
```diff
@@ -2,10 +2,10 @@
-       <changefreq>daily</changefreq>
-       <priority>0.8</priority>
-   </url>
+    <url>
```
I think previous indentation was correct.

wojtekmach replied:
it was mixed, 3 spaces in the first level and 4 in the 2nd level; now it's 4 levels everywhere and same as wikipedia article on sitemaps (at least the 1st code snippets) :-) I'll change it to 3 spaces everywhere since this is what sitemaps.org seems to be using

ericmj commented on test/fixtures/sitemap.xml:
```diff
@@ -3,9 +3,9 @@
+      <loc>https://hex.pm/packages/postgrex</loc>
+      <lastmod>2014-04-17T14:00:00Z</lastmod>
+      <changefreq>daily</changefreq>
+      <priority>0.8</priority>
```
Hard to see on github but it looks like this uses 3 spaces of indentation now?

wojtekmach replied:
yeah, it was 3 spaces in 1st level, and 4 spaces in 2nd level of indent, so I just unified it to be 3 spaces everywhere. I exchanged comments with @lexmag about these indentations and I really don't care which way it's gonna be, but I thought it should be unified. I can revert these changes too

ericmj replied:
Change it to either 2 or 4 spaces.

ericmj commented on web/models/package.ex:
```diff
@@ -122,8 +122,7 @@ defmodule HexWeb.Package do
          order_by: p.name,
-         where: not is_nil(p.docs_updated_at),
-         select: {p.name, p.docs_updated_at})
+         where: not is_nil(p.docs_updated_at))
```
We should keep the select because we want to keep the data transfer small. We are fetching every package in the database.

wojtekmach replied:
okay, I can revert this change, it's just I saw that hex.pm/sitemap.xml isn't doing a select and is fetching even more packages. I'll add this optimization to that controller along with ordering by name I guess since we do it here as well

ericmj replied:
Thanks

wojtekmach replied:
@ericmj I'm not sure what would be a good name for the 2nd function that's gonne be used by hex.pm/sitemap.xml, do you have any suggestions? Perhaps, instead `all_with_docs` we'd have something more specific, e.g.: `Package.packages_sitemap`, `Package.docs_sitemap`? Sorry for lots of questions, naming is hard :)

wojtekmach replied:
or something like this, perhaps?

```elixir
# web/models/sitemap.ex
defmodule Sitemap do
  def packages do
    from p in Package, ...
  end

  def docs do
    from p in Package, ...
  end
end
```

ericmj replied:
I like Package.packages_sitemap and Package.docs_sitemap.

wojtekmach commented:
Updated. I also renamed templates to be packages_sitemap.xml.eex and docs_sitemap.xml.eex to be consistent with the functions on Package. Thanks for your feedback everyone and let me know if anything is missing.

ericmj commented:
Thank you! :heart:
